website/docs/provider_reference.html.markdown

```diff
@@ -0,0 +1,95 @@
+---
+layout: "google"
+page_title: "google provider reference"
+sidebar_current: "docs-google-provider-reference"
+description: |-
+  The Google provider is used to configure your GCP project, location, and creds
+---
+
+# `google` provider reference
+The `google` provider block is used to configure default values for your GCP
+project and location (`zone` and `region`), and add your credentials.
+
+-> You can avoid using a provider block by using environment variables. Every
+field of the `google` provider is optional. If you want to share configs between
+environments and deploy to different projects, try it out!
+
+## Example Usage
+
+```hcl
+provider "google" {
+  credentials = "${file("account.json")}"
+  project     = "my-project-id"
+  region      = "us-central1"
+  zone        = "us-central1-c"
+}
+```
+
+## Configuration Reference
+
+The following keys can be used to configure the provider.
+
+* `credentials` - (Optional) The path or contents of a file that contains your
+  service account private key in JSON format. You can download your existing
+  [Google Cloud service account file]
+  from the Google Cloud Console, or you can create a new one from the same page.
+
+  Credentials can also be specified using any of the following environment
+  variables (listed in order of precedence):
+
+    * `GOOGLE_CREDENTIALS`
+    * `GOOGLE_CLOUD_KEYFILE_JSON`
+    * `GCLOUD_KEYFILE_JSON`
+
+  The [`GOOGLE_APPLICATION_CREDENTIALS`][adc]
+  environment variable can also contain the path of a file to obtain credentials
+  from.
```

Comment:
This should probably mention it's specifically for application default credentials, which are different from service account credentials, I believe.

Reply:
Based on https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually they look the same?

Reply:
I stand corrected, ignore me!